From ce64894d9496f9c065ecf090dfdd4793b3ab7b78 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Tue, 30 May 2023 11:23:57 +0800
Subject: [PATCH 93/93] pixman-renderer: Honor alpha channel for shadow

Use AR2 instead of XR24 for gbm format with alpha channel.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 libweston/pixman-renderer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libweston/pixman-renderer.c b/libweston/pixman-renderer.c
index 342a266..710e101 100644
--- a/libweston/pixman-renderer.c
+++ b/libweston/pixman-renderer.c
@@ -1495,7 +1495,7 @@ pixman_renderer_output_create(struct weston_output *output,
 		h = output->current_mode->height;
 
 		po->shadow_image =
-			pixman_image_create_bits_no_clear(PIXMAN_x8r8g8b8,
+			pixman_image_create_bits_no_clear(PIXMAN_a8r8g8b8,
 							  w, h, NULL, 0);
 
 		if (!po->shadow_image) {
-- 
2.20.1

